python/src/spark_rapids_ml/utils.py

```diff
@@ -267,6 +282,84 @@ def _concat_and_free(
     return concated
 
 
+def _concat_with_reserved_gpu_mem(
+    pdf_iter: Iterator[pd.DataFrame],
+    gpu_mem_ratio_for_data: float,
+    array_order: str,
+    logger: logging.Logger,
+) -> Tuple["cp.ndarray", Optional["cp.ndarray"], Optional[np.ndarray]]:
+    # TODO: check restriction on the array_order
+    # TODO: support multiple column and sparse matrix
+    # TODO: support row number
+    # TODO: support pytest -n 8
+
+    assert array_order == "C"
```

Comment:
These should be checked before call and default to previous logic if not the case.   Is it hopeless to make this zero copy approach work with 'F' ordering?

Reply:
It seems hopeless if taking submatrix directly from reserved gpu memory in 'F' order. 
Maybe there is a workaround. Thinking to manipulate values in place. Do we want to investigate that? If so will do it in next PR.